### PR TITLE
renamed DisconnectErrHandler to ConnErrHandler

### PR DESCRIPTION
**Pull request by mkorolyov**
As @kozlovic proposed this PR renames handler type to be more abstract so could be reused in future.

### CODE REVIEW

File: nats.go [hunk 1]
```diff
@@ -701,7 +701,7 @@ func DrainTimeout(t time.Duration) Option {
 }
 
 // DisconnectErrHandler is an Option to set the disconnected error handler.
-func DisconnectedErrHandler(cb DisconnectErrHandler) Option {
+func DisconnectedErrHandler(cb ConnErrHandler) Option {
```
[the remark this answers is not in the document]
Comment (kozlovic) [reply nats.go]: While at it, update `// DEPRECATED. use SetDisconnectErrHandler` to `// DEPRECATED: Use SetDisconnectErrHandler` line 875.

[the remark this answers is not in the document]
Comment (mkorolyov) [reply nats.go]: Sure, fixed docs, thanks for review!

File: nats.go [hunk 2]
```diff
@@ -701,14 +701,15 @@ func DrainTimeout(t time.Duration) Option {
...
-func DisconnectedErrHandler(cb ConnErrHandler) Option {
+func DisconnectErrHandler(cb ConnErrHandler) Option {
 	return func(o *Options) error {
 		o.DisconnectedErrCB = cb
 		return nil
 	}
 }
 
 // DisconnectHandler is an Option to set the disconnected handler.
+// DEPRECATED: Use Use DisconnectErrHandler.
```
Comment (kozlovic) [inline nats.go]: Sorry to be a pain: `Use Use` one too many ;-)

Comment (mkorolyov) [reply nats.go]: oh, my bad, cleaned

**Pull request merged**